core/spec.md

```diff
@@ -1951,9 +1951,15 @@ The following defines the specification-defined capabilities:
   A value of `false` indicates that the server MUST NOT perform any
   compatibility checking.
 
-  Attempts to change this value from `false` to `true` MUST fail if doing so
-  would result in any existing Version violating the `compatibility` rules
-  defined for the owning Resource.
+  Attempts to change this value from `false` to `true` SHOULD NOT result in
+  any validation of existing versions. Instead, compatibility will be
+  enforced for any new versions that are added.
+
+  This value can be set to `true` at any time, but can't be reset to`false`.
+  Once compatibility is enforced, it MUST remain enforced. Changing this
+  requires a new registry to be created as it's a fundamental change for
+  clients of the registry, which MUST be signaled accordingly.
```

Comment:
If we talked about this and I'm just forgetting, sorry....
why did you want this restriction? If changing this flag only impacts future versions, then what's the harm in turning it off? An impl can choose to ban this change if they want, and how they notify users, if it does change, is up to them... but from a spec perspective I'm not sure about banning it for all usecases.

Reply:
@duglin This is an angle that I don't think we fully agreed upon, so I left it in here purposefully to have the discussion. I can see it both ways:
- On the one hand, it's a very big restriction as you mentioned
- On the other hand, if compatibility strategies are set and the registry promises to enforce those strategies and later "changes its mind", this is a breaking change. Both producers and consumers are operating based on the assumption that compatibility is enforced which may cause ripple effects in the system. I'm leaning towards the stricter end but that may be due to my messaging-hat-bias. If the enforcement can be turned off at any time it loses much of its value imho. The way I see it is that this would likely mean that users will be much more careful in turning it on?

Reply:
My intent here was to remove the `enforcecompatibility` flag, as we agreed upon in previous meetings. This would mean that if you set the `compatibility` attribute of a Resource to a value, this is by default enforced (assuming the registry is able to enforce it). If you don't want the enforcement, this attribute should be set to `None`. 

However, thinking through this, I did have the following concern:
What if the registry, for whatever reason, cannot enforce compatibility, but still, you want to centrally register what the agreement was (even though it can't be enforced)? Now, you could only really do that by defining an extension that allows you to define that. Are we okay with that?

Reply:
It seems to me that we may really confuse people with fields that "sometimes" are verified/used by the server and "sometimes" not. And taken to this logical conclusion... if a server doesn't support any compat checking then the "compat" field should not be available at all. Likewise, setting it to a valid that the server can't support (even if it does support others) should not be a valid state. 

If we don't take a firm position on this then how does a client know when the server is enforcing something or not (ie. lying or not)?

If we don't do this then we wouldn't just need a "enforcecompat" flag for the client to enable/disable, but also some kind of "yes, I did check it" flag for the server to let clients know if it did it.